cdap-common/src/test/java/io/cdap/cdap/common/guice/ZKDiscoveryModuleTest.java

```diff
@@ -122,10 +122,10 @@ public void testProgramDiscovery() {
       // Register a service using the twill ZKClient. This is to simulate how a user Service program register
       ProgramId programId = NamespaceId.DEFAULT.app("app").service("service");
       String twillNamespace = injector.getInstance(CConfiguration.class).get(Constants.CFG_TWILL_ZK_NAMESPACE);
-      ZKClient twillZKClient = ZKClients.namespace(zkClient,
-                                                   twillNamespace + "/" + TwillAppNames.toTwillAppName(programId));
 
       String discoverableName = ServiceDiscoverable.getName(programId);
+      ZKClient twillZKClient = ZKClients.namespace(zkClient,
+                                                   twillNamespace + "/" + discoverableName);
```

Comment:
this test case fails without this change... the zkclient's namespace is impacting the discovery.